commons/src/test/java/com/powsybl/commons/extensions/ExtensionAdderProvidersTest.java

```diff
@@ -0,0 +1,355 @@
+        List<ExtensionAdderProvider> listProviders = Arrays.asList(
+                new SimpleExtensionAdderImplProvider(),
+                new SimpleExtensionAdderImpl2Provider(),
+                new GenericExtensionAdderImplProvider(),
+                new GenericExtensionAdderImpl2Provider());
+        ConcurrentMap<Pair<String, Class>, ExtensionAdderProvider> cache = new ConcurrentHashMap<>();
+        ConcurrentMap<String, List<ExtensionAdderProvider>> mapProviders = ExtensionAdderProviders.groupProvidersByName(listProviders);
+        ExtensionAdderProvider a = ExtensionAdderProviders.findCachedProvider("Default",
+                SimpleExtensionAdder.class, mapProviders, cache);
+        assertEquals("Default", a.getImplementationName());
+        assertEquals(SimpleExtensionAdderImplProvider.class, a.getClass());
+        assertEquals(1, cache.size());
+        ExtensionAdderProvider b = ExtensionAdderProviders.findCachedProvider("Custom",
+                SimpleExtensionAdder.class, mapProviders, cache);
+        assertEquals("Custom", b.getImplementationName());
+        assertEquals(SimpleExtensionAdderImpl2Provider.class, b.getClass());
+        assertEquals(2, cache.size());
+
+        // Do it again, it should work from the cache
+        assertEquals("Default", ExtensionAdderProviders.findCachedProvider("Default", SimpleExtensionAdder.class,
+                new ConcurrentHashMap<>(), cache).getImplementationName());
+        assertEquals("Custom", ExtensionAdderProviders.findCachedProvider("Custom", SimpleExtensionAdder.class,
+                                new ConcurrentHashMap<>(), cache)
+                        .getImplementationName());
+    }
+
+    @Test
+    public void testMissing() {
+        try {
+            ExtensionAdderProviders.findCachedProvider("Default",
+                    SimpleExtensionAdder.class, new ConcurrentHashMap<>(),
+                    new ConcurrentHashMap<>());
+            fail("Should throw Missing Provider exception");
+        } catch (PowsyblException e) {
+            assertTrue("Should throw Missing Provider exception",
+                    e.getMessage().contains("not found"));
+        }
+    }
+
+    @Test
+    public void testMissingAlternate() {
+        try {
+            List<ExtensionAdderProvider> listProviders = Arrays
+                    .asList(new SimpleExtensionAdderImplProvider());
+            ExtensionAdderProviders.findCachedProvider("Custom",
+                    SimpleExtensionAdder.class, new ConcurrentHashMap<>(),
+                    new ConcurrentHashMap<>());
+            fail("Should throw Missing Provider exception");
+        } catch (PowsyblException e) {
+            assertTrue("Should throw Missing Provider exception",
+                    e.getMessage().contains("not found"));
+        }
+    }
+
+    @Test
+    public void testMissingAlternate2() {
+        try {
+            List<ExtensionAdderProvider> listProviders = Arrays
```

Comment:
Unused variable?

Reply:
Good catch the test was not testing what I wanted to test